canonical-web-and-design/maas-ui
# Add fabric, space and vlan models to the store.

## Done

- Added actions + reducers + selectors for fabric + space + vlan models.
- Fetch fabrics, spaces, subnets and vlans when the KVM compose form is opened.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to a React KVM details page and select "Compose" from the action dropdown.
- Use Redux DevTools and check that fabrics, spaces, subnets and vlans are successfully loaded into state.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2045
